### PR TITLE
[device_info_plus_windows] Update ffi to 2.0.1

### PR DESCRIPTION
**Pull request by TabooSun**
## Description

Update ffi to 2.0.1.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by miquelbeltran** [comment]
So we had an internal discussion some days ago, if upgrading ffi was a major change or not, and we decided that it is the case because ffi 2 requires Dart 2.17.

Can you bump the package version a whole major number?

cc. @jpnurmi @vbuberen

File: packages/device_info_plus/device_info_plus/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 4.0.1
+
+- Update dependencies
+
```
Comment (miquelbeltran) [inline packages/device_info_plus/device_info_plus/CHANGELOG.md:4]: Should be 5.0.0

File: packages/device_info_plus/device_info_plus_windows/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 3.0.1
+
+- Update ffi to 2.0.1
+
```
Comment (miquelbeltran) [inline packages/device_info_plus/device_info_plus_windows/CHANGELOG.md:4]: Should be 4.0.0

**Review by vbuberen — CHANGES_REQUESTED** [review]
@miquelbeltran 
No, no major bump should happen here as `device_info_plus_windows` already had a major bump during update of ffi to 2.0.0 (where we needed to bump min Dart version) in one of recent PRs: https://github.com/fluttercommunity/plus_plugins/pull/918

Also, I don't thing that pubspec of `device_info_plus` should do a major bump as there is a minor change for windows platform only.

Thus, I believe we need to have `4.0.1` for `device_info_plus` and `3.0.1` for `device_info_plus_windows` in this PR.

File: packages/device_info_plus/device_info_plus/CHANGELOG.md (hunk 1 above)
Comment (miquelbeltran) [reply packages/device_info_plus/device_info_plus/CHANGELOG.md:4]: staying with 4.0.1 instead

File: packages/device_info_plus/device_info_plus_windows/CHANGELOG.md (hunk 2 above)
Comment (miquelbeltran) [reply packages/device_info_plus/device_info_plus_windows/CHANGELOG.md:4]: staying with 3.0.1 instead

**Comment by miquelbeltran** [comment]
Since new versions are back at 3.0.1 and 4.0.1, I am merging and releasing this today

**Pull request merged**